app/MacPorts/urls.py

```diff
@@ -35,7 +35,7 @@
     path('ports/filter/variant/', views.search_ports_in_variant, name='search_ports_in_variant'),
     path('ports/load_tickets/', views.tickets, name='trac_tickets'),
     path('ports/category/<slug:cat>/', views.categorylist, name='category_list'),
-    path('ports/variant/<slug:variant>/', views.variantlist, name='variant_list'),
+    url(r'^ports/variant/(?P<variant>[-a-zA-Z0-9_.]+)/$', views.variantlist, name='variant_list'),
```

Comment:
If this is intended to only match valid variant names, then the dash character shouldn't be in there.

Reply:
I don't know, having a bit flexible url might be fine?

Reply:
Honestly I wasn't even aware that dots became allowed in variant names (I was using underscores in the past, for example for `+perl5_28`, where a dot would make a lot more sense; I'm not sure if I want to change any of those now though: too much work for too little gain).

I kind of agree that having a bit more flexibility cannot hurt, I just want to point out that both `+` and `-` are explicitly forbidden even if we add more characters to the allowed set in the future, since these two characters usually represent delimiters between variants; in the same way as you cannot have `+` or `-` as valid character in variable names in `C` or `python`.

Again, flexibility cannot hurt, I wouldn't mind if this was part of some default allowed character set, but in case when you are explicitly listing the allowed characters, it might actively confuse future developers trying to figure out why a forbidden character was on the list in the first place. For this reason I would tend to agree with @jmroot.

My suggestion is to remove the dash and merge this. It might be nice to additionally (later) add some minor test case (broken before this commit and passing after), even though I'm not sure what that would be. I wouldn't wait for the test case, this needs to be merged ASAP as it's fixing a "seriously broken" behaviour.

Reply:
Okay, thank you. I have merged this after removing the dash.